core/store.go

```diff
@@ -63,3 +65,21 @@ func Del(k string) bool {
 	}
 	return false
 }
+
+func FilterVals(pattern string) []string {
+	var matchedValues []string
+	for key, value := range store {
+		val := value.Value.(string)
```

Comment:
Can we not use inherent encoding that we are maintaining instead of `reflect`?

Reply:
@manosriram this would not work. Use type encodings to check. Not this. Each Dice object holds the type and encoding information. `.(string)` is an expensive operation. we maintain type encoding information.